File: backend/src/main/java/com/EduPay/dto/PaymentCallback.java
package com.EduPay.dto;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PaymentCallback { // Changed to 'class'
    private String razorpayPaymentId;
    private String razorpayOrderId;
    private String razorpaySignature; // For verifying the callback authenticity
    private String status; // e.g., "success", "failed"
    private String errorMessage; // If payment failed

}

File: frontend/edupay/lib/models/payment_callback.dart
// lib/models/payment_callback.dart

import 'dart:convert';

class PaymentCallback {
  final String razorpayPaymentId;
  final String razorpayOrderId;
  final String razorpaySignature;
  final String status;
  final String? errorMessage;

  PaymentCallback({
    required this.razorpayPaymentId,
    required this.razorpayOrderId,
    required this.razorpaySignature,
    required this.status,
    this.errorMessage,
  });

  // Convert a PaymentCallback object to a JSON string
  String toJson() {
    return jsonEncode({
      'razorpayPaymentId': razorpayPaymentId,
      'razorpayOrderId': razorpayOrderId,
      'razorpaySignature': razorpaySignature,
      'status': status,
      'errorMessage': errorMessage,
    });
  }
}
